doc/source/security/selinux.rst

```diff
@@ -0,0 +1,507 @@
+
+2. To ``restorecon`` command applies the change. 
+
+    .. code-block:: bash
+
+        $ restorecon -R -v /var/lib/mysqlcustom
+
+3. Restart the mysqld service:
+
+    .. code-block:: bash
+
+        $ service mysqld start
+
+.. _selinux-custom-logs:
+
+Setting a Custom Log Location 
+-----------------------------
+
+If you do not use the default settings, SELinux, in enforcing mode, prevents access to the system. If you move your logs to a custom location in my.cnf:
+
+.. code-block:: text
+
+    log-error=/logs/mysqld.log
+
+Verify the logs with the following command:
+
+.. code-block:: bash
+
+    $ ls -1aZ
+    ...
+    drwxrwxrwx.   2 root root unconfined_u:object_r:default_t:s0    6 Dec  2 09:16 logs
+    ...
+
+Starting MySQL returns the following message:
+
+.. code-block:: bash
+
+    $ service mysql start
+    Redirecting to /bin/systemctl start mysql.service
+    Job for mysqld.service failed because the control process exited with error code.
+    See "systemctl status mysqld.service" and "journalctl -xe" for details.
+
+    $ journalctl -xe
+    ...
+    SELinux is preventing mysqld from write access to the directory logs.
+    ...
+
+The default SELinux policy allows mysqld to write logs into a directory tagged with ``var_log_t`, which is the ``/var/log``. You can solve the issue with the following methods:
+
+* Tag the ``/logs`` directory properly
+
+* Edit the SELinux policy to allow mysqld writing into directories tagged with the ``default_t``. 
+
+Tag the ``/logs`` directory is the recommended method. Run the following commands:
+
+.. code-block:: bash
+
+    $ semanage fcontext -a -t var_log_t /logs
+    $ restorecon -v /logs
+
+If you cannot change the ``/logs`` directory label, maybe other applications are using the same directory and those applications have their own rules, you must adjust the SELinux policy. Follow these steps:
+
+1. Create a local policy:
+
+    .. code-block:: bash
+
+        ausearch -c 'mysqld' --raw | audit2allow -M my-mysqld
+
+2. Two files are created: my-mysqld.te and my-mysqld.pp. The mysqld.te file is a human-readable policy definition. The my-mysqld.pp is the policy module that is loaded into the SELinux subsystem.
+
+    Open the my-myslqd.te file for editing:
+
+    .. code-block:: text
+
+        module my-mysqld 1.0;
+
+        require {
+            *type mysqld_t*;
+            type var_lib_t;
+            *type default_t*;
+            class file getattr;
+            *class dir write*;
+        }
+
+        #============= mysqld_t ==============
+        *allow mysqld_t default_t:dir write*;
+        allow mysqld_t var_lib_t:file getattr;
+
+    The policy contains rules for the custom data directory and the custom logs directory. We have set the proper labels for the data directory, and applying this autogenerated policy would loosen our hardening by allowing mysqld to access ``var_lib_t`` tags. 
```

Comment:
I would say that the generated policy module **may** include rules not related to what we are doing right now (log location), because it may contain some rules generated from our previous experiments (in this cases I have played with data directory and SELinux generated errors now converted into rules). 
So it is important to review this file, detect what is related to what we want to achieve, remove everything else.
That's why we use --start parameter below. Let' say that we played with data directory at 9:35:00 which caused SELinux journal logs generation.
Now it's 10:12:00 and we play with custom logs location. So we want to check the journal only from the time point when we started with this (let's say 10:00:00), as earlier events/errors/logs are not relevant.

Reply:
Fixed.